src/FakeItEasy/Expressions/ExpressionArgumentConstraintFactory.cs

```diff
@@ -106,5 +116,95 @@ private IArgumentConstraint CreateParamArrayConstraint(NewArrayExpression expres
 
             return new AggregateArgumentConstraint(result);
         }
+
+        private class ArgumentConstraintExpressionVisitor : ExpressionVisitor
+        {
+            private readonly List<Expression> constraintExpressions;

```

Comment:
Yes, that's an option too. We could even throw as soon as we find an invalid argument constraint. I can change it if you prefer.